embrace-android-features/src/test/java/io/embrace/android/embracesdk/internal/ndk/EmbraceNdkServiceTest.kt

```diff
@@ -364,58 +350,32 @@ internal class EmbraceNdkServiceTest {
     }
 
     @Test
-    fun `test checkForNativeCrash when a native crash was captured`() {
-        val crashFile: File = File.createTempFile("test", "test")
-        val errorFile: File = File.createTempFile("test", "test")
-        val mapFile: File = File.createTempFile("test", "test")
-
-        every { Uuid.getEmbUuid() } returns "unityId"
-
-        every { repository.errorFileForCrash(crashFile) } returns errorFile
-        every { repository.mapFileForCrash(crashFile) } returns mapFile
-
+    fun `test getLatestNativeCrash when a native crash was captured`() {
+        repository.addCrashFiles(
+            nativeCrashFile = File.createTempFile("test", "crash-test")
+        )
         every { delegate._getCrashReport(any()) } returns getNativeCrashRaw()
-        every { repository.sortNativeCrashes(false) } returns listOf(crashFile)
 
         configService.appFramework = AppFramework.UNITY
 
         initializeService()
         mockNativeSymbols()
 
-        val result = embraceNdkService.getLatestNativeCrash()
-        assertNotNull(result)
-
-        verify { embraceNdkService["getNativeCrashErrors"](any() as NativeCrashData, errorFile) }
-        verify(exactly = 1) { repository.sortNativeCrashes(false) }
-        verify(exactly = 1) { delegate._getCrashReport(any()) }
-        verify(exactly = 1) { repository.errorFileForCrash(crashFile) }
-        verify(exactly = 1) { repository.mapFileForCrash(crashFile) }
+        val result = checkNotNull(embraceNdkService.getLatestNativeCrash())
+        with(result) {
+            assertNotNull(crash)
+            assertNull(map)
```

Comment:
I didn't verify the map and errors because I would need to fake the serialized data in a file and that's just too much to add for at the moment. When we do the NDK refactor, we'll add the tests then.